File: EcuacionLineal_Java/metodoStatic/EcuacionLineal.java
package metodoStatic;

public class EcuacionLineal {

          public static double calcularValorDeX (float coeficienteA, float coeficienteB, float terminoIndependienteC, float coeficienteD, float coeficienteE, float terminoIndependienteF) {
              double determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD;
              double variableX = coeficienteE * terminoIndependienteC - coeficienteB * terminoIndependienteF / determinante;
              return variableX;
          }
          public static double calcuclarValorDeY (float coeficienteA, float coeficienteB, float terminoIndependienteC, float coeficienteD, float coeficienteE, float terminoIndependienteF) {
              double determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD;
              double variableY = coeficienteA * terminoIndependienteF - coeficienteD * terminoIndependienteC / determinante;
              return variableY;
          }
}

File: EcuacionLineal_Python/EcuacionLineal.py
from abc import abstractmethod
from abc import ABCMeta


class EcuacionLineal(ABCMeta):


    @staticmethod
    def calcularValorDeX(coeficienteA, coeficienteB, terminoIndependienteC,
                         coeficienteD, coeficienteE, terminoIndependienteF):
        determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD
        variableX = coeficienteE * terminoIndependienteC - coeficienteB * terminoIndependienteF / determinante
        return variableX

    @staticmethod
    def calcularValorDeY(coeficienteA, coeficienteB, terminoIndependienteC,
                         coeficienteD, coeficienteE, terminoIndependienteF):
        determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD
        variableY = coeficienteA * terminoIndependienteF - coeficienteD * terminoIndependienteC / determinante
        return variableY



